janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -878,20 +885,23 @@ public void restore(Map<String,Map<String, List<IndexEntry>>> documents, KeyInfo
             }
         }
         sr.setFrom(0);
-        if (query.hasLimit()) sr.setSize(query.getLimit());
-        else sr.setSize(maxResultsSize);
+        if (query.hasLimit()) {
+            sr.setSize(Math.min(query.getLimit(), nbResultByQuery));
+        } else {
+            sr.setSize(nbResultByQuery);
+        }
 
         ElasticSearchResponse response;
         try {
-            response = client.search(getIndexStoreName(query.getStore()), useMultitypeIndex ? query.getStore() : null, compat.createRequestBody(sr, NULL_PARAMETERS));
+            response = client.search(getIndexStoreName(query.getStore()), useMultitypeIndex ? query.getStore() : null, compat.createRequestBody(sr, NULL_PARAMETERS), sr.getSize() >= nbResultByQuery);
         } catch (IOException e) {
```

Comment:
Include the subsequent scrolls in the try-catch and add `UncheckedIOException` to the catch. Also can you split the long lines for clarity (maybe define indexName, typeName and ElasticSearchScroll iterator on separate lines)?